### DISCUSSION TITLE
Zsh/Bash autocompletion generator

### DISCUSSION THREAD

**Discussion by a-b** [erikhuda/thor#826] [Ideas]
It would be really handy to build script which will automatically generate autocompletion files.

**Comment by Zhomart** [comment]
This gem https://github.com/labocho/thor-zsh_completion works well. But it doesn't support options and subcommand arguments.

**Comment by tristanmorgan** [comment]
As an alternative to an "autocompletion generator", could Thor respond with completions when called with COMP_LINE and COMP_POINT? 
An example implemented and documented in a downstream project is [Autocomplete Design](https://github.com/servian/awskeyring/wiki/AutoComplete-Design)

**Comment by dorner** [comment]
That does sound like a better approach. PRs are welcome!

**Comment by tristanmorgan** [comment]
I'm not even going to create a PR (because my code is so bad) but here's a bit of it working.... [hacky auto-complete branch](https://github.com/tristanmorgan/thor/tree/autocomplete)
